include/cuco/detail/open_addressing/open_addressing_ref_impl.cuh

```diff
@@ -130,7 +130,30 @@ class open_addressing_ref_impl {
     probing_scheme_type const& probing_scheme,
     storage_ref_type storage_ref) noexcept
     : empty_slot_sentinel_{empty_slot_sentinel},
-      predicate_{this->extract_key(empty_slot_sentinel), predicate},
+      predicate_{
+        this->extract_key(empty_slot_sentinel), this->extract_key(empty_slot_sentinel), predicate},
+      probing_scheme_{probing_scheme},
+      storage_ref_{storage_ref}
+  {
+  }
+
+  /**
+   * @brief Constructs open_addressing_ref_impl.
+   *
+   * @param empty_slot_sentinel Sentinel indicating an empty slot
+   * @param erased_key_sentinel Sentinel indicating an erased key
+   * @param predicate Key equality binary callable
+   * @param probing_scheme Probing scheme
+   * @param storage_ref Non-owning ref of slot storage
+   */
+  __host__ __device__ explicit constexpr open_addressing_ref_impl(
+    value_type empty_slot_sentinel,
+    key_type erased_key_sentinel,
```

Comment:
Having a dedicated struct for all the sentinels will clean this up. Let's make it a future PR.

Reply:
[d2de184](https://github.com/NVIDIA/cuCollections/pull/365/commits/d2de184f27b9980c77af60e6691ebbcbda997255)